### ISSUE TITLE
Export is unstable

### ISSUE THREAD

**Issue by vmario89** [homeylab/bookstack-file-exporter#44]
Hi, this tool is great! But i got a bug report because its unstable and therefore not usable for production use case. Because if we have non-existing files, the whole export script stops:

`requests.exceptions.HTTPError: 404 Client Error: Not Found for url:`

i think we need some skip option + report generation about the export process

**Comment by vmario89** [comment]
```
.org'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
2024-10-29 15:24:59 [ERROR] Bookstack request failed with status code: 404 on url: https://docs.redacted.org/uploads/images/gallery/2024-10/redacted-netcup1.png
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/opt/bookstack-venv/lib/python3.12/site-packages/bookstack_file_exporter/__main__.py", line 16, in <module>
    main()
  File "/opt/bookstack-venv/lib/python3.12/site-packages/bookstack_file_exporter/__main__.py", line 12, in main
    run.exporter(args)
  File "/opt/bookstack-venv/lib/python3.12/site-packages/bookstack_file_exporter/run.py", line 45, in exporter
    archive.get_bookstack_exports(page_nodes)
  File "/opt/bookstack-venv/lib/python3.12/site-packages/bookstack_file_exporter/archiver/archiver.py", line 43, in get_bookstack_exports
    self._page_archiver.archive_pages(page_nodes)
  File "/opt/bookstack-venv/lib/python3.12/site-packages/bookstack_file_exporter/archiver/page_archiver.py", line 83, in archive_pages
    self.archive_page_assets("images", page.parent.file_path,
  File "/opt/bookstack-venv/lib/python3.12/site-packages/bookstack_file_exporter/archiver/page_archiver.py", line 132, in archive_page_assets
    asset_data = self.asset_archiver.get_asset_bytes(asset_type, asset_node.url)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bookstack-venv/lib/python3.12/site-packages/bookstack_file_exporter/archiver/asset_archiver.py", line 152, in get_asset_bytes
    asset_response: Response = common_util.http_get_request(
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/bookstack-venv/lib/python3.12/site-packages/bookstack_file_exporter/common/util.py", line 37, in http_get_request
    raise e
  File "/opt/bookstack-venv/lib/python3.12/site-packages/bookstack_file_exporter/common/util.py", line 31, in http_get_request
    response.raise_for_status()
  File "/opt/bookstack-venv/lib/python3.12/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://docs.redacted.org/uploads/images/gallery/2024-10/redacted-netcup1.png

```

**Comment by vmario89** [comment]
The script does **not** end with `log.info("Completed run")`

**Comment by pchang388** [comment]
hey @vmario89 thanks for opening the issue!

From what limited insight I have of your set up, it appears that in one of your page's contents, there is a reference to an image that no longer exists. I assume if you viewed that page the image wouldn't appear and the link wouldn't work (404). 

You could fix this by fixing or removing that stale/bad link in the page contents. But I believe you are asking for a feature to be able to skip an image/attachment that errors and log that information. That makes sense especially if the user doesn't have write access to fix/modify the page.

I'll take a look at implementing this in a few days when I get some free time off work. Thanks!

**Comment by vmario89** [comment]
Hi, thanks for your response! yes that was my request. Its okay when an image is missing, but the export should not stop for that completely

i already checked it out, but i cannot find the issue btw. The user i used to export is the admin will all permissions and the file is existing also. I have no idea where the error comes from

**Comment by pchang388** [comment]
That's interesting, so if you copy the link that has a 404 error, you see the image properly come up in your browser if you paste it there? That doesn't seem right but interested to see what happens when you do that. 

I guess it could be possible that image link in the Image API is wrong but the page has the correct one? Although doesn't sound likely. Anyways I am working on this when I have some time and should be pushing a new version soon

**Comment by pchang388** [comment]
For now, I pushed an image that contains the changes to skip an image or attachment if the API call fails. Could you give it a try using docker if possible? Try using the tag; `main` to test the latest changes, example:

```
# --user flag to override the uid/gid for created files. Set this to your uid/gid
docker run \
    -e LOG_LEVEL='info' \
    --user ${USER_ID}:${USER_GID} \
    -v $(pwd)/config.yml:/export/config/config.yml:ro \
    -v $(pwd)/bkps:/export/dump \
    homeylab/bookstack-file-exporter:latest
```

Hopefully this works around the issue for you for now, let me know how it goes, thanks!

**Comment by pchang388** [comment]
Fow now, since this is a minor change at surface level, I went ahead and pushed this new change. It is available in new version: `v1.3.0` and also `latest` tag (if using docker).

If the issue persists and you are still unable to skip images or attachments that have failed API calls, please feel free to re-open or open a new issue. Thank you again for taking the time to open this issue. 

If you ever have time to answer the questions I asked earlier, feel free to drop them in here too. Thanks!

**Comment by vmario89** [comment]
Hi @pchang388  many thanks for your investigations! i installd the recent version and now it works strongly like a horse :)
